validator/sawtooth_validator/server/core.py

```diff
@@ -203,7 +207,7 @@ def __init__(self,
             context_manager=context_manager,
             settings_view_factory=SettingsViewFactory(state_view_factory),
             scheduler_type=scheduler_type,
-            invalid_observers=[batch_tracker],
+            invalid_observers=[batch_tracker, batch_token_pool_updater],
```

Comment:
Can we move this observer to the caller of the scheduler? For the purposes of the batch_token_pool_updater, it seems like this could actually be just the block publisher since it is related to accepting batches to insert into the pending queue?